clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -811,6 +811,13 @@ public void onComplete(ClientResponse response) {
         log.trace("Sent produce request to {}: {}", nodeId, requestBuilder);
     }
 
+    /**
+     * Send serialization error as record-errors
+     */
+    public void sendSerializationError(String topic, int count) {
```

Comment:
If we don't have a plan to add more logic in this method, it should be OK to remove it and call `sender.sensors.recordErrors(topic, count)` directly.

Reply:
discussed offline. Sender.SenderMetrics is a private class and does not has a get function. Will stick to this method where we call sensor.recordErrors inside Sender.

Rename the sendSerializationError -> sendError to make it more generic (this function does not tie to serialization error only).